crates/cargo-util-schemas/src/manifest/mod.rs

```diff
@@ -776,12 +776,22 @@ impl<'de, P: Deserialize<'de> + Clone> de::Deserialize<'de> for TomlDependency<P
     where
         D: de::Deserializer<'de>,
     {
+        use serde::de::Error as _;
+        let expected = "a version string like \"0.9.8\" or a \
+                     detailed dependency like { version = \"0.9.8\" }";
         UntaggedEnumVisitor::new()
-            .expecting(
-                "a version string like \"0.9.8\" or a \
-                     detailed dependency like { version = \"0.9.8\" }",
-            )
+            .expecting(expected)
             .string(|value| Ok(TomlDependency::Simple(value.to_owned())))
+            .bool(|value| {
+                let err = format!(
+                    "invalid type: boolean `{value}`, expected {expected}\n\
+                    note: if you meant to use a workspace member, you can write\n \
+                      dep = {{ workspace = {value} }}\n\
```

Comment:
let's drop this option.

in our docs, we decided to standardize on the other form if they have nothing else to mention which is the case for `dep = true`